src/archive_writer.rs

```diff
@@ -302,45 +351,217 @@ fn write_symbol_table_header<W: Write + Seek>(
+    if aux_header.o_snloader() == 0 {
+        return MIN_BIG_ARCHIVE_MEM_DATA_ALIGN;
+    }
+
+    // The content of the loadable member file needs to be aligned at MAX(maximum
+    // alignment of .text, maximum alignment of .data) if there are both fields.
+    // If the desired alignment is > PAGESIZE, 32-bit members are aligned on a
+    // word boundary, while 64-bit members are aligned on a PAGESIZE(2^12=4096)
+    // boundary.
+    let log_2_of_align = u32::from(std::cmp::max(
+        aux_header.o_algntext(),
+        aux_header.o_algndata(),
+    ));
+    1 << (if log_2_of_align > LOG2_OF_AIXPAGE_SIZE {
+        log_2_of_max_align
+    } else {
+        log_2_of_align
+    })
+}
+
+// AIX big archives may contain shared object members. The AIX OS requires these
+// members to be aligned if they are 64-bit and recommends it for 32-bit
+// members. This ensures that when these members are loaded they are aligned in
+// memory.d
+fn get_member_alignment(sym_obj: &Option<SymbolicFile>) -> u32 {
+    use object::read::xcoff::FileHeader;
+
+    // If the desired alignment is > PAGESIZE, 32-bit members are aligned on a
+    // word boundary, while 64-bit members are aligned on a PAGESIZE boundary.
+    match sym_obj.as_ref() {
+        Some(SymbolicFile {
+            parsed: File::Xcoff64(xcoff_obj),
+            data,
+        }) => {
+            let mut offset = 0;
+            let Ok(header) = xcoff::FileHeader64::parse(*data, &mut offset) else {
+                return MIN_BIG_ARCHIVE_MEM_DATA_ALIGN;
+            };
+            let Ok(aux_header) = header.aux_header(*data, &mut offset) else {
+                return MIN_BIG_ARCHIVE_MEM_DATA_ALIGN;
+            };
+            get_aux_max_alignment(
+                xcoff_obj.raw_header().f_opthdr(),
+                aux_header,
+                LOG2_OF_AIXPAGE_SIZE,
+                offset_of!(object::xcoff::AuxHeader64, o_modtype)
+                    .try_into()
+                    .unwrap(),
+            )
+        }
+        Some(SymbolicFile {
+            parsed: File::Xcoff32(xcoff_obj),
+            data,
+        }) => {
+            let mut offset = 0;
+            let Ok(header) = object::xcoff::FileHeader32::parse(*data, &mut offset) else {
+                return MIN_BIG_ARCHIVE_MEM_DATA_ALIGN;
+            };
+            let Ok(aux_header) = header.aux_header(*data, &mut offset) else {
+                return MIN_BIG_ARCHIVE_MEM_DATA_ALIGN;
+            };
+            get_aux_max_alignment(
+                xcoff_obj.raw_header().f_opthdr(),
+                aux_header,
+                2,
+                offset_of!(object::xcoff::AuxHeader32, o_modtype)
+                    .try_into()
+                    .unwrap(),
+            )
+        }
+        _ => MIN_BIG_ARCHIVE_MEM_DATA_ALIGN,
+    }
+}
+
 fn write_symbol_table<W: Write + Seek>(
     w: &mut W,
     kind: ArchiveKind,
-    members: &[MemberData<'_>],
+    members: &[MemberData],
```

Comment:
```suggestion
    members: &[MemberData<'_>],
```

`MemberData` has a lifetime argument and the `rust_2018_idioms` lint group contains a lint which lints against hidden lifetimes.